staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/types.go

```diff
@@ -67,5 +69,24 @@ type Key struct {
 	Secret string `json:"secret"`
 }
 
+// GKMSConfig contains API configuration for Google KMS transformer.
+type GKMSConfig struct {
+	// projectID is the GCP project which hosts the key to be used. It is automatically set
+	// to your project if you are running Kubernetes on GKE/GCE.
+	// +optional
+	ProjectID string `json:"projectID,omitempty"`
+	// location is the location of the hosted key to be used. The default value is "global".
+	// +optional
+	Location string `json:"location,omitempty"`
+	// keyRing is the keyRing of the hosted key to be used. The default value is "google-kubernetes".
+	// +optional
+	KeyRing string `json:"keyRing,omitempty"`
+	// cryptoKey is the name of the key to be used for encryption of Data-Encryption-Keys.
```

Comment:
Also, there's an error which handles the case when CryptoKey or ProjectID is not assigned:
https://github.com/sakshamsharma/kubernetes/blob/312a7259141e789d889ff2f74c3bd86f51a539a3/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/kms/google.go#L57